packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -876,7 +876,13 @@ class _AndroidDevicePortForwarder extends DevicePortForwarder {
         process.throwException('adb did not report forwarded port');
       hostPort = int.tryParse(process.stdout) ?? (throw 'adb returned invalid port number:\n${process.stdout}');
     } else {
-      if (process.stdout.isNotEmpty)
+      // dantup: stdout may be empty or the port we asked it to forward,
+      // depending on some factors I do not understand. On MacOS it's always
+      // empty for Flutter, yet at the terminal it prints the port number
+      // only if it forwarded (nothing if it was already being forwarded).
+      // On CrOS it prints the port number even for Flutter, which results in a
+      // failure unless we allow the hostPort in the output here.
+      if (process.stdout.isNotEmpty && process.stdout.trim() != '$hostPort')
```

Comment:
I'm totally fine with this but we should phrase this as if it's by the whole team, not as if it's by one individual in the team. If someone really wants to track down who wrote it, they'll use git blame.